src/PhpSpreadsheet/Calculation/TextData/Concatenate.php

```diff
@@ -23,7 +25,18 @@ public static function CONCATENATE(...$args): string
         $aArgs = Functions::flattenArray($args);
 
         foreach ($aArgs as $arg) {
+            $value = Helpers::extractString($arg);
+            if (ErrorValue::isError($value)) {
+                $returnValue = $value;
+
+                break;
+            }
             $returnValue .= Helpers::extractString($arg);
+            if (mb_strlen($returnValue, 'UTF-8') > DataType::MAX_STRING_LENGTH) {
```

Comment:
Is it better to call the `Shared\StringHelper::countCharacters()` method for consistency?

Reply:
Yes, it is. I still haven't figured out where *all* the bodies are buried :-). Change is made. Note that a small change was necessary for countCharacters, allowing the specifying of the length parameter as null.